framework/src/mesh/SpiralAnnularMesh.C

```diff
@@ -0,0 +1,306 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SpiralAnnularMesh.h"
+
+#include "libmesh/face_quad4.h"
+#include "libmesh/face_tri3.h"
+
+registerMooseObject("MooseApp", SpiralAnnularMesh);
+
+template <>
+InputParameters
+validParams<SpiralAnnularMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredRangeCheckedParam<Real>(
+      "inner_radius", "inner_radius>0.", "The size of the inner circle.");
+  params.addRequiredRangeCheckedParam<Real>("outer_radius",
+                                            "outer_radius>0.",
+                                            "The size of the outer circle."
+                                            " Logically, it has to be greater than inner_radius");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "nodes_per_ring", "nodes_per_ring>0", "Number of nodes on each ring.");
+  params.addParam<bool>(
+      "my_second_order", false, "Generate mesh of TRI6 elements instead of TRI3 elements.");
```

Comment:
The plan is to eventually move the `second_order` parameter out of `SetupMeshAction` and into `MooseMesh` itself. In the meantime, this is a workaround that avoids our param clashing with the one in the Action.